src/CustomBeatmapLevelLoader.cpp

```diff
@@ -251,9 +251,11 @@ namespace RuntimeSongLoader::CustomBeatmapLevelLoader {
                                 [=] {
                                     try {
                                         self->_loadedBeatmapLevels->PutToCache(levelID, reinterpret_cast<IBeatmapLevel*>(customBeatmapLevel));
-                                    } catch (std::runtime_error const& e) {
+                                    } catch (std::exception const& e) {
                                         getLogger().Backtrace(20);
                                         LOG_ERROR("CustomBeatmapLevelLoader_GetBeatmapLevelAsync Failed to put (%s) to cache: %s!", static_cast<std::string>(levelID).c_str(), e.what());
+                                    } catch (...) {
+                                        LOG_ERROR("CustomBeatmapLevelLoader_GetBeatmapLevelAsync Failed to put (%s) to cache! unknown exception (not std::exception) with no known what() method!");

```

Comment:
please add `static_cast<std::string>(levelID).c_str()` so it logs the level id

Reply:
Yeah I just added it